src/turnkeyml/run/devices.py

```diff
@@ -72,3 +75,60 @@ def apply_default_runtime(device: str, runtime: Optional[str] = None):
         return DEVICE_RUNTIME_MAP[device][DEFAULT_RUNTIME]
     else:
         return runtime
+
+
+def select_runtime_and_sequence(
+    device: str, runtime: Optional[str], sequence: Optional[Sequence]
+) -> Tuple[str, str, Sequence]:
+    selected_runtime = apply_default_runtime(device, runtime)
+
+    # Validate device and runtime selections
+    if device not in SUPPORTED_DEVICES:
+        raise exp.ArgError(
+            f"Device argument '{device}' is not one of the available "
+            f"supported devices {SUPPORTED_DEVICES}\n"
+            f"You may need to check the spelling of '{device}', install a "
+            "plugin, or update the turnkeyml package."
+        )
+    else:
+        if selected_runtime not in DEVICE_RUNTIME_MAP[device]:
+            raise exp.ArgError(
+                f"Runtime argument '{selected_runtime}' is not one of the available "
+                f"runtimes supported for device '{device}': {DEVICE_RUNTIME_MAP[device]}\n"
+                f"You may need to check the spelling of '{selected_runtime}', install a "
+                "plugin, or update the turnkeyml package."
```

Comment:
While you are on it...
Maybe write this only once and reuse this for both error messages?

Reply:
This is just a copy-paste from a different file... but sure :)